core/templates/domain/topic/editable-topic-backend-api.service.ts

```diff
@@ -16,180 +16,194 @@
+      TopicDomainConstants.TOPIC_EDITOR_STORY_URL_TEMPLATE, {
+        topic_id: topicId
       });
-    };
 
-    var _fetchSubtopicPage = function(
-        topicId, subtopicId, successCallback, errorCallback) {
-      var subtopicPageDataUrl = UrlInterpolationService.interpolateUrl(
-        SUBTOPIC_PAGE_EDITOR_DATA_URL_TEMPLATE, {
-          topic_id: topicId,
-          subtopic_id: subtopicId.toString()
-        });
+    this.http.get(storiesDataUrl).toPromise().then((response: any) => {
+      let canonicalStorySummaries = cloneDeep(
+        response.canonical_story_summary_dicts);
+      if (successCallback) {
+        successCallback(canonicalStorySummaries);
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.error);
+      }
+    });
+  }
 
-      $http.get(subtopicPageDataUrl).then(function(response) {
-        var topic = angular.copy(response.data.subtopic_page);
-        if (successCallback) {
-          successCallback(topic);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _fetchSubtopicPage(
+      topicId: string, subtopicId: number,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    let subtopicPageDataUrl = this.urlInterpolation.interpolateUrl(
+      AppConstants.SUBTOPIC_PAGE_EDITOR_DATA_URL_TEMPLATE, {
+        topic_id: topicId,
+        subtopic_id: subtopicId.toString()
       });
-    };
 
-    var _deleteTopic = function(
-        topicId, successCallback, errorCallback) {
-      var topicDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_TOPIC_DATA_URL_TEMPLATE, {
-          topic_id: topicId
-        });
-      $http['delete'](topicDataUrl).then(function(response) {
-        if (successCallback) {
-          successCallback(response.status);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+    this.http.get(subtopicPageDataUrl).toPromise().then((response: any) => {
+      let topic = cloneDeep(response.subtopic_page);
+      if (successCallback) {
+        successCallback(topic);
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.error);
+      }
+    });
+  }
+
+  private _deleteTopic(topicId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    let topicDataUrl = this.urlInterpolation.interpolateUrl(
+      AppConstants.EDITABLE_TOPIC_DATA_URL_TEMPLATE, {
+        topic_id: topicId
       });
-    };
+    this.http['delete'](topicDataUrl).toPromise().then((response: any) => {
+      if (successCallback) {
+        successCallback(200);
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.body);
```

Comment:
Can we please make this errorResponse.error (as you've done above)? It's not a good idea to use error and body both. We will use error.

Reply:
sure @bansalnitish , done.